src/ERC3009.sol

```diff
@@ -0,0 +1,184 @@
+            ERC712.getDigest(
+                DOMAIN_SEPARATOR(),
+                keccak256(abi.encode(CANCEL_AUTHORIZATION_TYPEHASH, authorizer_, nonce_))
+            ),
+            v_,
+            r_,
+            s_
+        );
+
+        _authorizationStates[authorizer_][nonce_] = true;
+        emit AuthorizationCanceled(authorizer_, nonce_);
+    }
+
+    /******************************************************************************************************************\
+    |                                           Internal View/Pure Functions                                           |
+    \******************************************************************************************************************/
+
+    /**
+     * @notice Common transfer function used by `transferWithAuthorization` and `receiveWithAuthorization`.
+     * @param  typeHash_    Typehash of the transfer authorization
+     * @param  from_        Payer's address (Authorizer)
+     * @param  to_          Payee's address
+     * @param  value_       Amount to be transferred
+     * @param  validAfter_  The time after which this is valid (unix time)
+     * @param  validBefore_ The time before which this is valid (unix time)
+     * @param  nonce_       Unique nonce
+     * @param  v_           v of the signature
+     * @param  r_           r of the signature
+     * @param  s_           s of the signature
+     */
+    function _transferWithAuthorization(
+        bytes32 typeHash_,
+        address from_,
+        address to_,
+        uint256 value_,
+        uint256 validAfter_,
+        uint256 validBefore_,
+        bytes32 nonce_,
+        uint8 v_,
+        bytes32 r_,
+        bytes32 s_
+    ) internal {
+        if (block.timestamp < validAfter_) revert AuthorizationNotYetValid(block.timestamp, validAfter_);
+        if (block.timestamp > validBefore_) revert AuthorizationExpired(block.timestamp, validBefore_);
+        _revertIfAuthorizationAlreadyUsed(from_, nonce_);
+
+        ERC712.revertIfInvalidSignature(
+            from_,
+            ERC712.getDigest(
+                DOMAIN_SEPARATOR(),
+                keccak256(abi.encode(typeHash_, from_, to_, value_, validAfter_, validBefore_, nonce_))
+            ),
+            v_,
+            r_,
+            s_
+        );
+
+        _authorizationStates[from_][nonce_] = true;
+        emit AuthorizationUsed(from_, nonce_);
+
+        _transfer(from_, to_, value_);
+    }
+
+    /**
+     * @notice Reverts if the authorization is already used.
+     * @param  authorizer_ The authorizer's address
+     * @param  nonce_      The nonce of the authorization
+     */
+    function _revertIfAuthorizationAlreadyUsed(address authorizer_, bytes32 nonce_) internal view {
+        if (_authorizationStates[authorizer_][nonce_]) revert AuthorizationAlreadyUsed(authorizer_, nonce_);
+    }
+
+    /**
+     * @notice ERC20 transfer function that needs to ve overriden by the inheriting contract.
```

Comment:
```suggestion
     * @notice ERC20 transfer function that needs to be overridden by the inheriting contract.
```

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/7/commits/a27206542da32d6caa18a62cbfcbf772bf11a332